{
  "uid" : "d8eaebba2ddadf3",
  "name" : "1.3.3 Check response has key name",
  "fullName" : "tests.create_users_test.TestCreateUser#test_post_create_user_key_name_is_in_response",
  "historyId" : "52d7fb644553a589322a2631a2094c40",
  "time" : {
    "start" : 1726639124545,
    "stop" : 1726639124790,
    "duration" : 245
  },
  "status" : "skipped",
  "statusMessage" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
  "statusTrace" : "self = <tests.create_users_test.TestCreateUser object at 0x7fb8e1c08b50>\nkey_name = 'name'\n\n    @pytest.mark.positive\n    @pytest.mark.parametrize('key_name', data.LIST_KEY)\n    @pytest.mark.xfail(reason='Bag')\n    def test_post_create_user_key_name_is_in_response(self, key_name):\n        allure.dynamic.title(f'1.3.{self.data.LIST_KEY.index(key_name) + 1} Check response has key {key_name}')\n        headers = self.data.CREATE_USER\n        response = BaseRequests.post(url='api/users', headers=headers)\n>       Assertions.assert_key_name_is_in_response(response, key_name)\n\ntests/create_users_test.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = <Response [201]>, key_name = 'name'\n\n    @staticmethod\n    def assert_key_name_is_in_response(response: Response, key_name):\n        \"\"\"Check if the required key is in the response\"\"\"\n        json_value = response.json()\n        actual_key = json_value.get(key_name)\n        with allure.step(f'The key {actual_key} exist'):\n>           assert actual_key is not None, f'The key {actual_key} does not exist'\nE           AssertionError: The key None does not exist\n\nbase/assertions.py:36: AssertionError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1726639123517,
      "stop" : 1726639123532,
      "duration" : 15
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false,
    "stepsCount" : 0
  } ],
  "testStage" : {
    "status" : "skipped",
    "statusMessage" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
    "statusTrace" : "self = <tests.create_users_test.TestCreateUser object at 0x7fb8e1c08b50>\nkey_name = 'name'\n\n    @pytest.mark.positive\n    @pytest.mark.parametrize('key_name', data.LIST_KEY)\n    @pytest.mark.xfail(reason='Bag')\n    def test_post_create_user_key_name_is_in_response(self, key_name):\n        allure.dynamic.title(f'1.3.{self.data.LIST_KEY.index(key_name) + 1} Check response has key {key_name}')\n        headers = self.data.CREATE_USER\n        response = BaseRequests.post(url='api/users', headers=headers)\n>       Assertions.assert_key_name_is_in_response(response, key_name)\n\ntests/create_users_test.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = <Response [201]>, key_name = 'name'\n\n    @staticmethod\n    def assert_key_name_is_in_response(response: Response, key_name):\n        \"\"\"Check if the required key is in the response\"\"\"\n        json_value = response.json()\n        actual_key = json_value.get(key_name)\n        with allure.step(f'The key {actual_key} exist'):\n>           assert actual_key is not None, f'The key {actual_key} does not exist'\nE           AssertionError: The key None does not exist\n\nbase/assertions.py:36: AssertionError",
    "steps" : [ {
      "name" : "The key None exist",
      "time" : {
        "start" : 1726639124790,
        "stop" : 1726639124790,
        "duration" : 0
      },
      "status" : "failed",
      "statusMessage" : "AssertionError: The key None does not exist\n",
      "statusTrace" : "  File \"/home/runner/work/reqres/reqres/base/assertions.py\", line 36, in assert_key_name_is_in_response\n    assert actual_key is not None, f'The key {actual_key} does not exist'\n           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true,
      "stepsCount" : 0
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true,
    "stepsCount" : 1
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Create User"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.xfail(reason='Bag')"
  }, {
    "name" : "tag",
    "value" : "positive"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "create_users_test"
  }, {
    "name" : "subSuite",
    "value" : "TestCreateUser"
  }, {
    "name" : "host",
    "value" : "fv-az530-206"
  }, {
    "name" : "thread",
    "value" : "2272-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.create_users_test"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "key_name",
    "value" : "'name'"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 0,
        "skipped" : 11,
        "passed" : 0,
        "unknown" : 0,
        "total" : 11
      },
      "items" : [ {
        "uid" : "2c76ef2ee0ffc025",
        "reportUrl" : "https://Nat754.github.io/reqres/13//#testresult/2c76ef2ee0ffc025",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708432773648,
          "stop" : 1708432774013,
          "duration" : 365
        }
      }, {
        "uid" : "d346f1370b2162d5",
        "reportUrl" : "https://Nat754.github.io/reqres/12//#testresult/d346f1370b2162d5",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708427532473,
          "stop" : 1708427532725,
          "duration" : 252
        }
      }, {
        "uid" : "2dc9d02e052e2b5b",
        "reportUrl" : "https://Nat754.github.io/reqres/11//#testresult/2dc9d02e052e2b5b",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708425058919,
          "stop" : 1708425059170,
          "duration" : 251
        }
      }, {
        "uid" : "de02d621f322c734",
        "reportUrl" : "https://Nat754.github.io/reqres/10//#testresult/de02d621f322c734",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708420916716,
          "stop" : 1708420916973,
          "duration" : 257
        }
      }, {
        "uid" : "6ba011891d3b3fa9",
        "reportUrl" : "https://Nat754.github.io/reqres/9//#testresult/6ba011891d3b3fa9",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708420824763,
          "stop" : 1708420826044,
          "duration" : 1281
        }
      }, {
        "uid" : "6dfbcd8ec7ae15bb",
        "reportUrl" : "https://Nat754.github.io/reqres/8//#testresult/6dfbcd8ec7ae15bb",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708420745964,
          "stop" : 1708420746211,
          "duration" : 247
        }
      }, {
        "uid" : "a9ad727f797d9ebd",
        "reportUrl" : "https://Nat754.github.io/reqres/7//#testresult/a9ad727f797d9ebd",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708418351086,
          "stop" : 1708418351461,
          "duration" : 375
        }
      }, {
        "uid" : "5940bb9defe5080d",
        "reportUrl" : "https://Nat754.github.io/reqres/6//#testresult/5940bb9defe5080d",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708412184455,
          "stop" : 1708412184700,
          "duration" : 245
        }
      }, {
        "uid" : "92dfdbab903ae166",
        "reportUrl" : "https://Nat754.github.io/reqres/5//#testresult/92dfdbab903ae166",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708388621669,
          "stop" : 1708388621910,
          "duration" : 241
        }
      }, {
        "uid" : "ee6cac6eb643f774",
        "reportUrl" : "https://Nat754.github.io/reqres/4//#testresult/ee6cac6eb643f774",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708385916243,
          "stop" : 1708385916626,
          "duration" : 383
        }
      } ]
    },
    "tags" : [ "@pytest.mark.xfail(reason='Bag')", "positive" ]
  },
  "source" : "d8eaebba2ddadf3.json",
  "parameterValues" : [ "'name'" ]
}